cmd/node_agent/na/nodeagent.go

```diff
@@ -78,43 +79,98 @@ type platformSpecificRequest interface {
 	IsProperPlatform() bool
 }
 
+// CAGrpcClient is for implementing the GRPC client to talk to CA.
+type CAGrpcClient interface {
+	// Send CSR to the CA and gets the response or error.
+	SendCSR(*string, []grpc.DialOption, *pb.Request) (*pb.Response, error)
+}
+
+// CAGrpcClientImpl is a implementation of GRPC client to talk to CA.
+type CAGrpcClientImpl struct {
+}
+
+// SendCSR sends CSR to CA through GRPC.
+func (c *CAGrpcClientImpl) SendCSR(address *string, options []grpc.DialOption, req *pb.Request) (*pb.Response, error) {
```

Comment:
Are we expecting the address and options to change? If not shall we set them as members of `CAGrpcClientImpl` struct?